File: sonarqube-companion-rest/src/main/java/net/lipecki/sqcompanion/group/GroupSummary.java
package net.lipecki.sqcompanion.group;

import lombok.Builder;
import lombok.Data;
import net.lipecki.sqcompanion.health.HealthStatus;
import net.lipecki.sqcompanion.violations.Violations;

/**
 * @author gregorry
 */
@Data
@Builder
public class GroupSummary {

	private String uuid;
	private String name;
	private HealthStatus healthStatus;
	private Violations violations;

}

File: sonarqube-companion-frontend/src/app/group/group-summary.ts
import {HealthStatus} from '../health/health-status';
import {Violations} from '../violations/violations';

export class GroupSummary {

  uuid: string;
  name: string;
  healthStatus: HealthStatus;
  violations: Violations;

  constructor(data) {
    this.uuid = data.uuid;
    this.name = data.name;
    this.healthStatus = HealthStatus[data.healthStatus] as HealthStatus;
    this.violations = new Violations(data.violations || {});
  }

}
